src/libraries/System.Runtime/tests/System/HashCodeTests.cs

```diff
@@ -121,6 +122,75 @@ public static void HashCode_Add_NullEqualityComparer()
         Assert.Equal(expected.ToHashCode(), hc.ToHashCode());
     }
 
+    [Fact]
+    public static void HashCode_Add_Span_AllValuesHashed()
+    {
+        static void Populate(ref HashCode hc, int length)
+        {
+            byte[] data = Enumerable.Range(1, length).Select(i => (byte)i).ToArray();
+            hc.Add((ReadOnlySpan<byte>)data);
+        }
+
+        HashCode hc = default;
+        Populate(ref hc, 1);
+
+        int last = hc.ToHashCode();
+        for (int i = 2; i < 100; i++)
+        {
+            hc = default;
+            Populate(ref hc, i);
+            int value = hc.ToHashCode();
+            Assert.NotEqual(last, value);
```

Comment:
I get that it's very unlikely, but the hashcode is not guaranteed to be different just because new data has been ingested.

Reply:
I know, but I don't know how else to validate this.  Suggestions?

Reply:
We could use reflection to read the `HashCode._length` field, but I'm sure that is more likely to introduce test breaks than a theoretical collision concern.

Reply:
We should expect this unit test to fail after ~20 million runs. Honestly that doesn't seem like a great sin considering some of our other tests. If we see flakiness here we can probably address it as needed?

Reply:
Changed it to use reflection.